suites/apis/dbgapTest.js

```diff
@@ -0,0 +1,280 @@
+    ).to.equal(fence.props.googleBucketInfo.test.fileContents);
+
+    chai.expect(phst000179s3FileContents,
+      `User ${users.mainAcct.username} WITHOUT access COULD create s3 signed urls ` +
+      'and read file for a record in unauthorized dbGaP project phs000179'
+    ).to.not.equal(fence.props.awsBucketInfo.cdis_presigned_url_test.testdata);
+    chai.expect(phst000179gsFileContents,
+      `User ${users.mainAcct.username} WITHOUT access COULD create gs signed urls ` +
+      'and read file for a record in unauthorized dbGaP project phs000179'
+    ).to.not.equal(fence.props.googleBucketInfo.QA.fileContents);
+});
+
+Scenario('dbGaP + user.yaml Sync: ensure combined access @dbgapSyncing',
+  async (fence, indexd, users, files) => {
+    console.log('Running usersync job and adding dbgap sync to yaml sync');
+    bash.runJob('usersync', args='ADD_DBGAP true FORCE true');
+
+    // users.mainAcct has access to phs000178 through dbgap
+    console.log('Use mainAcct to create s3 signed URL for file phs000178')
+    const signedUrls3phs000178Res = await fence.do.createSignedUrl(
+      indexed_files.phs000178File.did, ['protocol=s3'],
+      users.mainAcct.accessTokenHeader
+    );
+
+    // users.mainAcct has access to QA through user.yaml
+    console.log('Use mainAcct to create s3 signed URL for file QA')
+    const signedUrlsQARes = await fence.do.createSignedUrl(
+      indexed_files.QAFile.did, ['protocol=s3'],
+      users.mainAcct.accessTokenHeader
+    );
+
+    let phs000178s3FileContents = await fence.do.getFileFromSignedUrlRes(
+      signedUrls3phs000178Res);
+    let QAFileContents = await fence.do.getFileFromSignedUrlRes(
+      signedUrlsQARes);
+
+    chai.expect(phs000178s3FileContents,
+      `User ${users.mainAcct.username} with access could NOT create s3 signed urls ` +
+      'and read file for a record in authorized dbGaP project phs000178'
+    ).to.equal(fence.props.awsBucketInfo.cdis_presigned_url_test.testdata);
+    chai.expect(QAFileContents,
+      `User ${users.mainAcct.username} with access could NOT create s3 signed urls ` +
+      'and read file for a record in authorized program QA'
+    ).to.equal(fence.props.awsBucketInfo.cdis_presigned_url_test.testdata);
+});
+
+Scenario('dbGaP + user.yaml Sync (from prev test): ensure user cannot create/update/delete dbGaP indexd records @dbgapSyncing',
+  async (fence, indexd, users, files) => {
+    console.log('populating guids for indexd records to attempt to create...');
+    // populate new GUIDs per test
+    const dbgapFooBarFileGuid = uuid.v4().toString();
+    const dbgapDeleteMe = uuid.v4().toString();
+
+    new_dbgap_records.fooBarFile.did = dbgapFooBarFileGuid;
+    new_dbgap_records.deleteMe.did = dbgapDeleteMe;
+
+    // create
+    const dbgap_create_success = await indexd.do.addFileIndices(
+        Object.values(new_dbgap_records), users.user2.accessTokenHeader
+    );
+
+    // read to test creation
```

Comment:
the tests in the `centralizedAuth` test suite check record creation the same way, and i guess trying to read the record can be a good extra check, but you can start with:
```
const ok = await indexd.do.addFileIndices();
chai.expect(ok).to.be.true;
```

Reply:
`addFileIndices` creates 2 records and you only test reading one
you could just add one of them (i just noticed we don't even have an indexd function to add a singe record haha)

Reply:
oh, this specific test is checking for a `404` to make sure it did _not_ get created. THe next section `force creation of records` does just check if the response is true

Reply:
oh right 😆